client/rest-high-level/src/main/java/org/elasticsearch/client/ml/inference/TrainedModelInput.java

```diff
@@ -37,27 +39,39 @@
     @SuppressWarnings("unchecked")
     public static final ConstructingObjectParser<TrainedModelInput, Void> PARSER = new ConstructingObjectParser<>(NAME,
         true,
-        a -> new TrainedModelInput((List<String>) a[0]));
+        a -> new TrainedModelInput((List<InputObject>) a[0]));
 
     static {
-        PARSER.declareStringArray(ConstructingObjectParser.constructorArg(), FIELD_NAMES);
+        PARSER.declareFieldArray(ConstructingObjectParser.constructorArg(),
```

Comment:
I see your point, but changing the field name to `input_fields` would mean that the 7.6 HLRC would get a `null` value for `field_names`. Which is worse?

Since the APIs are experimental, I vote towards the "loudest" error.